packages/wonder-blocks-search-field/src/components/__tests__/search-field.test.js

```diff
@@ -0,0 +1,181 @@
+import userEvent from "@testing-library/user-event";
+
+import SearchField from "../search-field.js";
+
+describe("SearchField", () => {
+    test("value is updated when text is entered into the field", () => {
+        // Arrange
+        const SearchFieldWrapper = () => {
+            const [value, setValue] = React.useState("");
+            return (
+                <SearchField
+                    id="sf-id"
+                    value={value}
+                    onChange={setValue}
+                    testId="search-field-test"
+                />
+            );
+        };
+
+        render(<SearchFieldWrapper />);
+
+        // Act
+        // Type something so the clear button appears.
+        const searchField = screen.getByTestId("search-field-test");
+        searchField.focus();
+        userEvent.paste(searchField, "a");
+
+        // Assert
+        expect(searchField).toHaveValue("a");
+    });
+    test("aria props are passed down", () => {
+        // Arrange
+
+        // Act
+        render(
+            <SearchField
+                id="sf-id"
+                value=""
+                onChange={() => {}}
+                testId="search-field-test"
+                aria-label="some-aria-label"
+            />,
+        );
+
+        // Assert
+        const input = screen.getByRole("textbox");
+        expect(input).toHaveAttribute("aria-label", "some-aria-label");
+    });
+
+    test("receives focus on click", () => {
+        // Arrange
+        render(
+            <SearchField
+                id="sf-id"
+                value=""
+                onChange={() => {}}
+                testId="search-field-test"
+            />,
+        );
+
+        // Act
+        const searchField = screen.getByTestId("search-field-test");
+        userEvent.click(searchField);
+
+        // Assert
+        expect(searchField).toHaveFocus();
+    });
+
+    test("receives focus on tab", () => {
+        // Arrange
+        render(
+            <SearchField
+                id="sf-id"
+                value=""
+                onChange={() => {}}
+                testId="search-field-test"
+            />,
+        );
+
+        // Act
+        userEvent.tab();
+
+        // Assert
+        const searchField = screen.getByTestId("search-field-test");
+        expect(searchField).toHaveFocus();
+    });
+
+    test("loses focus if tabbed off", () => {
+        // Arrange
+        render(
+            <SearchField
+                id="sf-id"
+                value=""
+                onChange={() => {}}
+                testId="search-field-test"
+            />,
+        );
+
+        // Act
+        // focus
+        userEvent.tab();
+        // blur
+        userEvent.tab();
+
+        // Assert
+        const searchField = screen.getByTestId("search-field-test");
+        expect(searchField).not.toHaveFocus();
+    });
+
+    test("does not have clear icon by default", () => {
```

Comment:
Another test that would be good including is verifying that `clearAriaLabel` is applied to the dismiss icon.